ReleaseProcessAutomation.IntegrationTests/GitBackedTests.cs

```diff
@@ -20,18 +20,75 @@
 using System.IO;
 using NUnit.Framework;
 
-namespace ReleaseProcessAutomation.Tests;
+namespace ReleaseProcessAutomation.IntegrationTests;
 
 public class GitBackedTests
 {
-  protected string PreviousWorkingDirectory;
-  protected string RepositoryPath;
+  protected static void ExecuteGitCommand (string argument)
+  {
+    using var command = Process.Start("git", argument);
+    command.WaitForExit();
+  }
+
+  protected static string ExecuteGitCommandWithOutput (string argument)
+  {
+    var psi = new ProcessStartInfo("git", argument);
+    psi.UseShellExecute = false;
+    psi.RedirectStandardOutput = true;
+    psi.RedirectStandardError = true;
+
+    using var command = Process.Start(psi);
+    command.WaitForExit();
+    if (command.ExitCode != 0)
+    {
+      var error = command.StandardError.ReadToEnd();
+    }
+
+    var output = command.StandardOutput.ReadToEnd();
+
+    return output;
+  }
+
+  protected void AddCommit (string s = "")
+  {
+    var random = new Random().Next();
+    ExecuteGitCommand($"commit --allow-empty -m {random}");
+  }
+
+  protected void AddRandomFile (string directoryPath)
+  {
+    var random = new Random().Next();
+
+    var path = Path.Join(directoryPath, random.ToString());
+
+    using var fs = File.Create(path);
+    fs.Dispose();
```

Comment:
Either `using` or `.Dispose()` is needed and I think that `.Dispose()` is clearer.

```suggestion
    var fs = File.Create(path);
    fs.Dispose();
```